Add support to show close button for active tab only.

kovdu opened the pull request:
This MR introduces `activeOnly ` for the `showCloseButton` theme option causing the close button only to appear on the active tab.

This is more or less following the approach explained here https://github.com/orgs/microsoft/projects/686/views/2?pane=issue&itemId=19775774 which indeed just works 😄 .

You notice when switching theme the close buttons is back on all tabs again as well.

Closes #13672

I didn't check specific unit tests for this. I hope by making this MR the pipeline will show if I broke something. Or just let me know if you want me to add something specific for this.

zadjii-msft reviewed (approved):
Yep that's literally how I'd do it 😝 Thanks!

carlos-zamora commented on src/cascadia/TerminalApp/TerminalPage.cpp:
```diff
@@ -3109,47 +3109,60 @@ namespace winrt::TerminalApp::implementation
         // Begin Theme handling
         _updateThemeColors();
 
+        const auto tab{ _GetFocusedTab() };
+        _updateTabsCloseButton(tab);
```
nit: no extra var necessary
```suggestion
        _updateTabsCloseButton(_GetFocusedTab());
```

carlos-zamora reviewed (approved):
Love it! Thanks for the contribution! Please be sure to...
- [ ] update the [schema](https://github.com/microsoft/terminal/blob/main/doc/cascadia/profiles.schema.json)
- [ ] update the [docs](https://github.com/MicrosoftDocs/terminal)

Verified that this works if you change the `showCloseButton` setting via hot reload.

carlos-zamora commented:
## Demo

Small demo of the result:
![active_only](https://user-images.githubusercontent.com/13299273/234409243-2d16d29b-61b3-4626-bbe8-e433808ad9d6.gif)


NOTE: removed the demo from the PR body so that the commit history is a bit cleaner

carlos-zamora commented:
@kovdu I'm not going to merge this yet to give you some time to apply/respond to my comments, but the PR is good to go! Again, thanks for the contribution!

kovdu commented:
Thanks for the feedback @carlos-zamora .

Fixed your remarks, updated `profiles.schema.json` to reflect the new option and created an MR for updating the documentation as well: https://github.com/MicrosoftDocs/terminal/pull/660. Not sure what the process is to get that one merged though.

Also verified the new value is correctly applied when hot reloading the settings file.
